app.py

```diff
@@ -1,45 +1,22 @@
 import os
-from pathlib import Path
 
 import chainlit as cl
 from chainlit.input_widget import Slider, TextInput
-from chainlit.playground.config import add_llm_provider
-from chainlit.playground.providers.langchain import LangchainGenericProvider
-from langchain.chains import ConversationalRetrievalChain
+from langchain import hub
+from langchain.callbacks.base import BaseCallbackHandler
 from langchain.globals import set_llm_cache
-from langchain.memory import ConversationBufferMemory
-from langchain_community.cache import SQLiteCache
-from langchain_community.chat_message_histories import ChatMessageHistory
-from langchain_community.chat_models import ChatOllama
-from langchain_community.embeddings import HuggingFaceEmbeddings
-from langchain_community.vectorstores import Chroma
+from langchain.schema.runnable.config import RunnableConfig
+from langchain_chroma import Chroma
+from langchain_community.cache import InMemoryCache
 from langchain_core.callbacks import CallbackManager, StreamingStdOutCallbackHandler
+from langchain_core.output_parsers import StrOutputParser
+from langchain_core.runnables import RunnablePassthrough
+from langchain_huggingface import HuggingFaceEmbeddings
+from langchain_ollama import ChatOllama
 
 from embed import load_config, parse_args
 
 
-def setup_memory():
```

Comment:
Removing memory because it's been migrated to another function that can be used with the system memory, however that function is not yet compatible with my runnable, so I will have to file an issue to add that support back.